addons/point_of_sale/static/src/js/models.js

```diff
@@ -1562,12 +1562,6 @@ exports.PosModel = Backbone.Model.extend({
      * fetch them to be added on the loaded products.
      */
     async _addProducts(ids){
-        await this.rpc({
-            model: 'product.product',
-            method: 'write',
-            args: [ids, {'available_in_pos': true}],
-            context: this.session.user_context,
-        });
```

Comment:
You don't need to remove this. You're changing a desired behavior used in other places (where this method is called). 
Also, since you're getting the ids of the product only available in the pos, this rpc won't change anything in your fix

Reply:
Yes, you're totally right. The idea was to avoid useless write RPC because, as you said, this won't change anything in my case, but I didn't figured out the implications this can have on other call (really stupid and dangerous mistake indeed).

Anyway, for performances reasons, shouldn't we make something to make sure we only write if necessary ? Or maybe this is out of the scope of a bugfix and I should simply let this part of code as it ?

Reply:
I'll reply to you when I get more info about the issue. I've heard that making the available_in_pos through the database search feature was a desired behavior